src/dispatch/plugins/dispatch_slack/case/interactive.py

```diff
@@ -1332,6 +1333,126 @@ def handle_escalation_submission_event(
     )
 
 
+@app.action(
+    CaseNotificationActions.migrate,
+    middleware=[button_context_middleware, db_middleware, user_middleware],
+)
+def create_channel_button_click(
+    ack: Ack,
+    body: dict,
+    client: WebClient,
+    context: BoltContext,
+    form_data: dict,
+    db_session: Session,
+):
+    ack()
+    case = case_service.get(db_session=db_session, case_id=context["subject"].id)
+    case.dedicated_channel = True
+    db_session.commit()
+
+    blocks = [
+        Section(text="Migrate the thread conversation to a dedicated channel?"),
+        Context(elements=[MarkdownText(text="This action will remove the case from this thread.")]),
+    ]
+
+    modal = Modal(
+        title="Create Case Channel",
+        blocks=blocks,
+        submit="Create Channel",
+        close="Close",
+        callback_id=CaseMigrateActions.submit,
+        private_metadata=context["subject"].json(),
+    ).build()
+    client.views_open(trigger_id=body["trigger_id"], view=modal)
+
+
+def ack_handle_create_channel_event(ack: Ack, case: Case) -> None:
+    """Handles the case channel creation event."""
+    msg = (
+        "The case already has a dedicated channel. No actions will be performed."
+        if case.has_channel
+        else "Creating a dedicated case channel..."
+    )
+
+    modal = Modal(
+        title="Creating Case Channel",
+        close="Close",
+        blocks=[Section(text=msg)],
+    ).build()
+
+    ack(response_action="update", view=modal)
+
+
+@app.view(
+    CaseMigrateActions.submit,
+    middleware=[
+        action_context_middleware,
+        # user_middleware,
+        # button_context_middleware,
+        db_middleware,
+        # modal_submit_middleware,
+    ],
+)
+def handle_create_channel_event(
+    ack: Ack,
+    body: dict,
+    client: WebClient,
+    context: BoltContext,
+    db_session: Session,
+    form_data: dict,
+    user: DispatchUser,
+):
+    """Handles the escalation submission event."""
+    case = case_service.get(db_session=db_session, case_id=context["subject"].id)
+    ack_handle_create_channel_event(ack=ack, case=case)
+
+    case.dedicated_channel = True
+    db_session.commit()
+
+    msg = (
+        "Creating a dedicated case channel..."
+        if not case.has_channel
+        else "The case already has a dedicated channel. No actions will be performed."
+    )
+
+    modal = Modal(
+        title="Creating  Case  Channel",
+        close="Close",
+        blocks=[Section(text=msg)],
+    ).build()
+
+    result = client.views_update(
+        view_id=body["view"]["id"],
+        trigger_id=body["trigger_id"],
+        view=modal,
+    )
+
+    channel_id = case.conversation.channel_id
+    thread_id = case.conversation.thread_id
+
+    participant_emails = [participant.individual.email for participant in case.participants]
```

Comment:
dupe of:

unrelated thought : i think we do this in quite a few places, potentially we could make this a property in the case/incident/participants model

Reply:
Done.